index.d.ts

```diff
@@ -1,4 +1,6 @@
 declare namespace mapObject {
+	type DeepOptionFunction = (sourceKey: string) => boolean;
```

Comment:
`Function` is obvious from the deep. The name should instead describe what it does.

Reply:
It's also missing a doc comments. The docs in the readme should be in sync with index.d.ts.

Reply:
Okay. I would suggest instead `DeepKeyFilter` as a name, would you confirm that?

Concerning the readme I used the specifiaction of the mapper function as a "template" to document the option; will commit an update then for `DeepKeyFilter`

Reply:
cf. https://github.com/sindresorhus/map-obj/pull/24/commits/fa9b55f6938e9ce05cf2b546e3317d71f3dd98c6